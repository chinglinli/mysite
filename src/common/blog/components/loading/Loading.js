import React from 'react'
import classNames from 'classnames'

import styles from './Loading.scss'

export default class Loading extends React.PureComponent {
  render () {
    return (
      <div className={styles.root}>
        <div className={classNames(styles.loadingImage, styles.twister)}/>
        <div className={styles.message}>正在努力讀取中…</div>
      </div>
    )
  }
}
