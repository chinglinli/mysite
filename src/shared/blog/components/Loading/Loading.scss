.root {
  margin-top: 10%;
  display: flex;
  flex-direction: column;
  align-items: center;

  .loadingImage {
      display: inline-block;

      width: 90px;
      height: 90px;
      background: url('./img/loading.jpg') no-repeat center;
      background-size: 90px 90px;

      border: 0.1em solid #FFFFFF;
      box-shadow: 0 1px 4px rgba(0, 0, 0, 0.15);
    }

  .message {
    margin-top: 2rem;
    font-size: 2rem;
    text-align: center;
  }
}


.twister {
  animation-name: twister;
  animation-duration: 5s;
  animation-iteration-count: infinite;
  z-index: 1000;
}

@keyframes twister {

  0% {
    transform: rotateZ(0deg);
  }

  20% {
    transform: rotateZ(360deg);
  }

  40% {
    transform: rotateZ(720deg);
  }

  60% {
    transform: rotateZ(1080deg);
  }

  80% {
    transform: rotateZ(1440deg);
  }
}
