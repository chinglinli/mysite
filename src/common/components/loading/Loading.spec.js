import React from 'react'
import renderer from 'react-test-renderer'

import Loading from './Loading'

test('It should render Loading component correctly', () => {
  const component = renderer.create(
    <Loading />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})