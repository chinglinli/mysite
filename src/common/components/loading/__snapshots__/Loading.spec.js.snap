// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render Loading component correctly 1`] = `
<div
  className="root"
>
  <div
    className="loadingImage twister"
  />
  <div
    className="message"
  >
    正在努力讀取中…
  </div>
</div>
`;
